Server/datalayer/leave.datalayer.js

```diff
@@ -1,13 +1,43 @@
+const mongoose = require('mongoose');
+const leavesModel = require('../models/leave.model');
 
-/*
- * @class leavesDatalayer
- */
-class leavesDatalayer {
+class leavesDataLayer {
 
-    /**
-    * Initializes the datalayer
-    */
+    static getLeavesByUserId (userId) {
+        return leavesModel.find({'userId': mongoose.Types.ObjectId(userId)}).exec().catch((err) => {
+            throw new Error (`Could not retrieve leaves. Error message - ${err.message}`);
+        });
+    }
 
-}
+    static createLeave (reqObj) {
+        let {endDate, reason, startDate} = reqObj;
+        const newLeave = new leavesModel({
+            startDate: startDate,
+            endDate: endDate,
+            reason: reason,
+            userId: mongoose.Types.ObjectId(reqObj.userId)
```

Comment:
We should extract userId also as done in first line

Reply:
Done.